caffe2/core/storage.h

```diff
@@ -140,13 +140,14 @@ class CAFFE2_API StorageImpl : public c10::intrusive_ptr_target {
     } else {
       data_ptr_.reset(src, d);
     }
-    capacity_ = capacity;
+    assert(data_type_.id() == data_type.id());
```

Comment:
This is not the idiomatic way to assert in the Caffe2 codebase; they use CAFFE_ENFORCE instead.